dali/operators/reader/loader/video_loader.cc

```diff
@@ -211,19 +212,77 @@ typename std::enable_if<!std::numeric_limits<T>::is_integer, bool>::type
         || std::abs(x-y) < std::numeric_limits<T>::min();
 }
 
+static int read_packet(void *opaque, uint8_t *buf, int buf_size) {
+  OpenFile::VideoFileData *file_data = static_cast<OpenFile::VideoFileData*>(opaque);
+
+  // open file if needed
+  if (!file_data->file_stream) {
+    file_data->file_stream = fopen(file_data->filename.c_str(),"r");
+    if (!file_data->file_stream) {
+      DALI_FAIL(std::string("Could not open file ") + file_data->filename);
+    }
+    fseek(file_data->file_stream, file_data->file_position, SEEK_SET);
+
+  }
+  return fread(buf, 1, buf_size, file_data->file_stream);
+}
+
+int64_t seek_file(void *opaque, int64_t offset, int whence) {
+  OpenFile::VideoFileData *file_data = static_cast<OpenFile::VideoFileData*>(opaque);
+
+  // open file if needed
+  if (!file_data->file_stream) {
+    file_data->file_stream = fopen(file_data->filename.c_str(),"r");
+    if (!file_data->file_stream) {
+      DALI_FAIL(std::string("Could not open file ") + file_data->filename);
+    }
+    fseek(file_data->file_stream, file_data->file_position, SEEK_SET);
+  }
+  int ret = -1;
+  switch (whence) {
+    case SEEK_SET:
+    case SEEK_CUR:
+    case SEEK_END:
+      ret = fseek(file_data->file_stream, offset, whence);
+      break;
+    default:
+      break;
+  }
+  if (ret == 0) {
+    return ftell(file_data->file_stream);
+  } else {
+    return ret;
```

Comment:
This is ambiguous - the standard doesn't say that the value returned on error is negative. Perhaps we should throw?

Reply:
It is not the standard. It is what FFmpeg expects, and expects some error code which is < 0.

Reply:
I meant that `ftell` does not guarantee a negative value on error - just nonzero. You probably should return -1 or sth.